### ISSUE TITLE
mongoose 5.4.20 throws ValidationError on save with multi-level nested schemas

### ISSUE THREAD

**Issue by cdraper-CS** [Automattic/mongoose#7650]
**Do you want to request a *feature* or report a *bug*?**

A bug.

**What is the current behavior?**

Calling save on an existing document with two levels of nested schema and the lowest level having at least two fields throws a validation error in mongoose 5.4.20.

**If the current behavior is a bug, please provide the steps to reproduce.**

The following works with mongoose 5.4.19. With mongoose 5.4.20 it throws a validation error on the save of `Error: ValidationError: anObject.aString: Path `aString` is required., anObject: Validation failed: aString: Path `aString` is required.`

```
const mongoose = require('mongoose');

const { Schema } = mongoose;

const MyObjectSchema = new Schema({
    aString: { required: true, type: String },
    aNumber: { required: true, type: Number },
});

const MyModelSchema = new Schema({
    name: { type: 'String', required: true },
    anObject: { type: MyObjectSchema, required: true },
});

const MyModel = mongoose.model('mymodel', MyModelSchema);

(async () => {
    try {
        const params = { aString: 'foo', aNumber: 1 };
        await mongoose.connect('mongodb://localhost:27017/issue');
        console.log('Doing');
        await MyModel.findOneAndUpdate({ name: 'test' }, { name: 'test', anObject: params }, { upsert: true });
        const myModel = await MyModel.findOne({ name: 'test' });
        myModel.anObject = params;
        await myModel.save();
        console.log('Done');
    } catch (err) {
       console.error(`Error: ${err}`);
    }
})();
```
Note: if I change the 'required' to false, it does not throw an Error but instead quietly throws the aString field away.

**What is the expected behavior?**

Expected behavior is as 5.4.19 - no validation error

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

node v8.12.0
mongoose 5.4.20
mongodb server 3.4.10, 3.6.11
mongodb package 3.1.13

**Comment by jglcanalcar** [comment]
This may be related with https://github.com/Automattic/mongoose/issues/7639

**Comment by juanmait** [comment]
related to this one: https://github.com/Automattic/mongoose/issues/7660
and this one: https://github.com/Automattic/mongoose/issues/7656

**Comment by vkarpov15** [comment]
I can confirm this was fixed in 5.5.1 along with the fix for #7660. Please upgrade to 5.5.1 to fix this issue :+1:
